src/main/java/ch/bildspur/realsense/RealSenseCamera.java

```diff
@@ -620,9 +633,13 @@ public PVector getProjectedPoint(int x, int y) {
     public short[][] getDepthData() {
         checkRunning();
 
-        DepthFrame frame = frames.getDepthFrame();
-        depthStream.updateDepthData(frame);
-        frame.release();
+        if (frames != null) {
```

Comment:
So this means that the depthStream won't be updated if the frame is null and we get the same depth data back? I guess for stability this is good but it could be misleading for the user. But I think for now this is ok.

Reply:
That is my understanding. I think that I've been running into these errors because I'm running Realsense updates in a separate thread. Receiving an occasional frame of cached data seems better to me than potential multithreading-related errors.